modules/standard/Map.chpl

```diff
@@ -284,8 +294,14 @@ module Map {
                this map.
     */
     iter items() const ref {
-      for key in myKeys {
-        yield (key, vals[key]);
+      if chpl_nilable {
+        for key in myKeys {
+          yield (key, vals[key]);
+        }
+      } else {
+        for key in myKeys {
+          yield (key, vals[key]: valType);
```

Comment:
Are we sure that this doesn't just create a reference to a temporary value? Maybe we can add a test where we modify the reference after use.